src/sympc/session/session.py

```diff
@@ -185,41 +185,98 @@ def get_protocol(self) -> Protocol:
         """
         return self.protocol
 
-    def przs_generate_random_share(
+    def _generate_random_share(
         self,
         shape: Union[tuple, torch.Size],
-    ) -> Any:
-        """Generate a random share using the two generators hold by a party.
+    ) -> List[torch.Tensor]:
+        """Generate random tensor share for the given shape.
 
         Args:
             shape (Union[tuple, torch.Size]): Shape for the share.
 
         Returns:
-            Any: ShareTensor
-
+            List[torch.Tensor] : shares generated by the generators.
         """
-        from sympc.tensor import ShareTensor
-
         gen0, gen1 = self.przs_generators
```

Comment:
Maybe: Could have these as `gen1`, `gen2` to be consistent with the rest of the changes in the PR

Reply:
Thats a good suggestion.Sure , I would change it :+1: